### ISSUE TITLE
app_radgw - "Undefined symbol" when loading .rgwx plugins

### ISSUE THREAD

**Issue by atari83** [freeDiameter/freeDiameter#8]
Hello !

I've built the latest version (1.6) on FreeBSD 13 (llvm/clang 11) successfully and with no issues.
Then I've configured freeDiameterd to use app_radgw extension (with auth.rgwx plugin). Well I received some error when I started the daemon: 

20:30:11   DBG   Loading of plugin '/usr/local/lib/freeDiameter/auth.rgwx' failed: /usr/local/lib/freeDiameter/auth.rgwx: Undefined symbol "rgw_clients_get_origin"

The only solution that I found was to make change to CMakeLists.txt:

**Old:**

```
### Authentication, Authorization messages translation.
FD_OPTION_PLUGIN(RGWX_AUTH "Build Authentication & Authorization RADIUS translation plugin? (RFC2865, RFC3579)" ON)
        IF (RGWX_AUTH)
           RGWX_ADD_PLUGIN(auth ${RG_COMMON_HEADER} rgwx_auth.c)
        ENDIF (RGWX_AUTH)

```

**New:**

```
### Authentication, Authorization messages translation.
FD_OPTION_PLUGIN(RGWX_AUTH "Build Authentication & Authorization RADIUS translation plugin? (RFC2865, RFC3579)" ON)
        IF (RGWX_AUTH)
           RGWX_ADD_PLUGIN(auth ${RG_COMMON_HEADER} rgwx_auth.c ${RGW_DEFAULT_SRC})
        ENDIF (RGWX_AUTH)

```

I know it is not good solution, but at least it worked. Could anybody please take look at this and find better solution ?

Thanks

**Comment by lukem** [comment]
This may be related to my change on 2020-03-30 of extensions/CMakeLists.txt to hide symbols by default.

I'll look at either:
1. adding the required sources to each of the app_radgw plugins, or
2. change app_radgw to export all of its symbols.

**Comment by lukem** [comment]
I've fixed this on the proposed branch in commit 9157ae6, using item 1, which required adding some of the sources to some of the .rwgx plugins.

I also improved the errors in startup to log at ERROR logging instead of DBG, as inspired by your description, in commit ede40af.

**Comment by atari83** [comment]
Thank you Luke :)
